massa-api/src/public.rs

```diff
@@ -945,6 +957,53 @@ impl MassaRpcServer for API<Public> {
             .collect())
     }
 
+    async fn get_address_datastore_keys(
+        &self,
+        arg: Vec<GetAddressDatastoreKeysRequest>,
+    ) -> RpcResult<Vec<GetAddressDatastoreKeysResponse>> {
+        let requests: Vec<ExecutionQueryRequestItem> = arg
+            .into_iter()
+            .map(|request| {
+                from_get_address_datastore_keys(
+                    request,
+                    self.0.api_settings.max_datastore_keys_queries,
+                )
+            })
+            .collect();
+        // let mut result: Vec<Vec<Vec<u8>>> = Vec::with_capacity(requests.len());
+        let mut result2: Vec<GetAddressDatastoreKeysResponse> = Vec::with_capacity(requests.len());
+
+        let response = self
+            .0
+            .execution_controller
+            .query_state(ExecutionQueryRequest { requests });
+
+        for response_item in response.responses {
+            match response_item {
+                Ok(item) => match item {
+                    ExecutionQueryResponseItem::AddressDatastoreKeys(keys, address, is_final) => {
+                        result2.push(GetAddressDatastoreKeysResponse {
+                            address,
+                            is_final,
+                            keys: keys.into_iter().collect(),
+                        });
+                        // result.push(keys.into_iter().collect());
+                    }
+                    _ => {
+                        return Err(
+                            ApiError::ExecutionError("unexpected response".to_string()).into()
```

Comment:
note: this is also triggered when keys from a non-existing address are queried, which is not unexpected but could be quite common

Reply:
same comment elsewhere and in the grpc

Reply:
same comment elsewhere

Reply:
Here we match for the type of enum `ExecutionQueryResponseItem`, we can't have anything else than AddressDatastoreKeys type. So i return an Error "in case if" 
For NotFound error, it will be triggered at line 996